### ISSUE TITLE
TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by adamreisnz** [wycats/handlebars.js#1250]
I get the following error when using `{{#unless @last}}<br>{{/unless}}` in an `each` block inside of a table structure, e.g.

``` html
<table class="AddressTable">
    <tr>
      <td>
        <strong>Factuuradres</strong><br>
        {{customer.name}}<br>
        {{#each customer.address.parts as |part|}}
          {{part}}{{#unless @last}}<br>{{/unless}}
        {{/each}}
      </td>
      <td>
        {{#unless order.isPickUp}}
        <strong>Bezorgadres</strong><br>
        {{#if order.delivery.companyName}}
          {{order.delivery.companyName}}<br>
        {{/if}}
        {{order.delivery.name}}<br>
        {{#each order.delivery.address.parts as |part|}}
          {{part}}{{#unless @last}}<br>{{/unless}}
        {{/each}}
        {{/unless}}
      </td>
    </tr>
  </table>
```

When I move the whole `unless` block out of the table, it works fine. If I wrap the `unless` block in an additional `<p>` tag inside of the '<td>`, it also works. What's happening here?

```
TypeError: Cannot read property '0' of undefined
    at eval (eval at createFunctionContext (node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:6:62)
    at prog (node_modules/handlebars/dist/cjs/handlebars/runtime.js:219:12)
    at execIteration (node_modules/handlebars/dist/cjs/handlebars/helpers/each.js:51:19)
    at Object.<anonymous> (node_modules/handlebars/dist/cjs/handlebars/helpers/each.js:61:13)
    at eval (eval at createFunctionContext (node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:10:31)
    at Object.prog [as inverse] (node_modules/handlebars/dist/cjs/handlebars/runtime.js:219:12)
    at Object.<anonymous> (node_modules/handlebars/dist/cjs/handlebars/helpers/if.js:17:22)
    at Object.<anonymous> (node_modules/handlebars/dist/cjs/handlebars/helpers/if.js:24:35)
    at Object.eval (eval at createFunctionContext (node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:20:33)
    at main (node_modules/handlebars/dist/cjs/handlebars/runtime.js:173:32)
    at ret (node_modules/handlebars/dist/cjs/handlebars/runtime.js:176:12)
    at ret (node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:525:21)

```

This is with 4.0.5 on Node 6.5.0

**Comment by adambuczynski** [comment]
It also barks when I move it all to a `<div>` structure as follows:

```html
<div class="AddressTable">
    <p>
      <strong>Factuuradres</strong><br>
      {{customer.name}}<br>
      {{#each customer.address.parts as |part|}}
        {{part}}{{#unless @last}}<br>{{/unless}}
      {{/each}}
    </p>
    <p>
      {{#unless order.isPickUp}}
      <strong>Bezorgadres</strong><br>
      {{#if order.delivery.companyName}}
        {{order.delivery.companyName}}<br>
      {{/if}}
      {{order.delivery.name}}<br>
      {{#each order.delivery.address.parts as |part|}}
        {{part}}{{#unless @last}}<br>{{/unless}}
      {{/each}}
      {{/unless}}
    </p>
  </div>
```

**Comment by yermokhin** [comment]
I have the same issue in my complex template. And I have found that in most cases this issue occur when I have the "each" variable (in you case it is "part", named the same in at least 2 each cycles. Just try to change "part"  in the second for each to something unique like "part2" or "item".

**Comment by adambuczynski** [comment]
Interesting, I'll give it a try, thanks.

**Comment by olehyermokhin** [comment]
Looks like that https://github.com/wycats/handlebars.js/pull/1187/ this pull request fix the problem!
I have updated handlebar to the latest version http://builds.handlebarsjs.com.s3.amazonaws.com/handlebars-latest.js and the issue is not reproducible any more

**Comment by nknapp** [comment]
@adamreisnz @olehyermokhin Since #1187 is part of 4.0.6, this should be fixed. Can you verify that? Would be great if we could close the issue.

**Comment by adamreisnz** [comment]
Have re-tested with the HTML structure as I had posted above and it seems to work ok now with 4.0.6 🎉

**Comment by nknapp** [comment]
Great, thank you.

Am 31. Dezember 2016 02:23:21 MEZ, schrieb Adam Reis <notifications@github.com>:

**Comment by olehyermokhin** [comment]
I'm sorry for late feedback. I can also confirm that it works with 4.0.6. Thanks a lot!

**Comment by jneuendorf** [comment]
The latest release is v4.0.5 (20 Nov 16). Why is v4.0.6 behind? If I do the upgrade from v4.0.5 to v4.0.6 are there any missing features or different bugs?

**Comment by nknapp** [comment]
There actually is a 4.0.6, but it's not available from all source. And there *are* some issues with it.
Have a look at #1312. It contains a list of bugs that would be fixed with 4.0.7.

I'm currently talking to @wycats in order to get the necessary access rights to perform a release myself. Once I have them (and the necessary time), I want to publish 4.0.7
